E3SM-Project/scream
# Add new SCREAM_TEST_LEVEL concept

Current SCREAM_TEST_LEVELS:
```
0) "AT" : stands for autotester. This is the default. Test at this level
   are expected to always pass and not take a long time to run.
1) "NIGHTLY" : These are tests that run nightly. They can take longer and
   fails are possible but should be addressed in a reasonable time frame.
2) "EXPERIMENTAL" : These tests are not run by any automated test process
   and do not appear on our dashboard. This is for testing future features
   that are not fully developed yet.
```

```
Usage:
  Command line:
    cmake ... -DSCREAM_TEST_LEVEL=NIGHTLY ...

  Cmake:
    Making a level-restricted test:
      CreateUnitTest(${test_name} ${test_src} MINIMUM_TEST_LEVEL ${SCREAM_TEST_LEVEL_NIGHTLY})

    Making general level-restricted cmake logic:
      # Code to run only if test level is at least nightly
      if (SCREAM_TEST_LEVEL GREATER_EQUAL SCREAM_TEST_LEVEL_NIGHTLY)
        # do nightly/experimental stuff
      endif()

  test-all-scream:
    ./test-all-scream --test-level=nightly ...
```

Note that, in test-all-scream, test-level is not pinned to a specific build type (dbg, opt, etc)
but instead can be set for all test types.

Change list:
1) Add --test-level option to test-all-scream
2) Add support for MINIMUM_TEST_LEVEL option to Scream's CreateUnitTest/CreateUnitTestFromExec Cmake macro
  - This is the first Scream-only option (EKAT is unaware of test level) for our Cmake test API
3) Add tests for this new feature in scripts-tests and corresponding cmake tests in meta-tests/test_level_check
4) Remove SCREAM_ENABLE_EXPERIMENTAL cmake option since that functionaly is now available in this new concept
